assets/js/modules/adsense/util/data-mock.js

```diff
@@ -0,0 +1,221 @@
+					max: lastValue ? Math.ceil( lastValue * ( 1 + delta ) ) : 500,
+				} );
+
+				values.push( newValue.toFixed( 2 ) );
+				generateMetricValues.memory[ metric ] = newValue;
+				break;
+			}
+			case METRIC_RATIO: {
+				const newValue = faker.random.float( {
+					min: lastValue ? lastValue * ( 1 - delta ) : faker.random.number( { min: .1, max: .9 } ),
+					max: lastValue ? Math.min( lastValue * ( 1 + delta ), 1 ) : 1,
+				} );
+
+				values.push( newValue.toFixed( 2 ) );
+				generateMetricValues.memory[ metric ] = newValue;
+				break;
+			}
+			default:
+				values.push( '' );
+				break;
+		}
+	}
+
+	return values;
+}
+
+/**
+ * Generates and returns dimension values.
+ *
+ * @since n.e.x.t
+ *
+ * @param {string}         date       The current date.
+ * @param {Array.<string>} dimensions Dimensions list.
+ * @return {Array.<string>} Array of dimension values.
+ */
+function generateDimensionValues( date, dimensions ) {
+	const values = [];
+
+	for ( const dimension of dimensions ) {
+		switch ( dimension.toUpperCase() ) {
+			case 'DATE':
+				values.push( date );
+				break;
+		}
+	}
+
+	return values;
+}
+
+/**
+ * Generates mock data for AdSense reports.
+ *
+ * @since n.e.x.t
+ *
+ * @param {Object} args Report options.
+ * @return {Array.<Object>} An array with generated report.
+ */
+export function getAdSenseMockResponse( args ) {
+	const originalSeedValue = faker.seedValue;
+	const argsHash = parseInt(
+		md5( JSON.stringify( args ) ).substring( 0, 10 ),
+		16,
+	);
+
+	// We set seed for every data mock to make sure that the same arguments get the same report data.
+	// It means that everyone will have the same report data and will see the same widgets in the storybook.
+	// This approach gives us additional flexibility to control randomness on a per widget basis.
+	if ( ! Number.isNaN( argsHash ) ) {
+		faker.seed( argsHash );
+	}
+
+	const metrics = ( Array.isArray( args.metrics ) ? args.metrics : [ args.metrics ] ).filter( ( metric ) => !! metric );
+	const dimensions = ( Array.isArray( args.dimensions ) ? args.dimensions : [ args.dimensions ] ).filter( ( dimension ) => !! dimension );
+
+	const data = {
+		kind: 'adsense#report',
+		startDate: args.startDate,
+		endDate: args.endDate,
+		totalMatchedRows: '0',
+		headers: generateHeaders( metrics, dimensions ),
+		totals: [],
+		averages: [],
+		rows: [],
+	};
+
+	const startDate = new Date( args.startDate );
+	const endDate = new Date( args.endDate );
+	const dayInMilliseconds = 24 * 60 * 60 * 1000;
+	const totalDays = 1 + ( ( endDate - startDate ) / dayInMilliseconds ); // +1 to include the endDate into the dates range.
+
+	// This is the list of operations that we will apply to the range (array) of numbers.
+	const ops = [
+		// Converts range number to a date string.
+		map( ( item ) => ( new Date( startDate.getTime() + ( dayInMilliseconds * item ) ) ).toISOString().split( 'T' )[ 0 ] ),
```

Comment:
It looks like this is creating a new date incremented by `item` days from the start. How about something like this instead?

```js
new Date( startDate ).setDate( startDate.getDate() + item );
```

Also for formatting, if this is our date string, we should use our `getDateString` utility instead.

Reply:
Updated. :+1: